### ISSUE TITLE
Vagrant insecure key replacement to support Mac OS X

### ISSUE THREAD

**Issue by koldrid** [mitchellh/vagrant#5513]
Right now Vagrant is not able to replace the insecure vagrant SSH certificates with randomly generated ones when spinning up the Guest OS on Mac OS 10.9 and 10.10. I have not tried with earlier versions of Mac OS but I will guess its the same behavior.

**Comment by neuroticnerd** [comment]
@koldrid its a little hard to tell from your description, but is https://github.com/mitchellh/vagrant/issues/5492 what you're experiencing?

**Comment by koldrid** [comment]
Sorry about that @neuroticnerd. I just posted this since the error message said to open a issue to get this OS "Mac OS" to support replacing the SSH Keypair with the randomly generated one. Below is the output of the error. Once I get this error I am able to run the command again and I am able to SSH into the host.

```
-----> Starting Kitchen (v1.3.1)
-----> Creating <default-mac-os-10-10-2>...
       Bringing machine 'default' up with 'vmware_fusion' provider...
       ==> default: Cloning VMware VM: 'mac_os_10_10_2_vmware'. This can take some time...
       ==> default: Verifying vmnet devices are healthy...
       ==> default: Preparing network adapters...
       ==> default: Starting the VMware VM...
       ==> default: Waiting for machine to boot. This may take a few minutes...
           default: SSH address: 192.168.163.214:22
           default: SSH username: vagrant
           default: SSH auth method: private key
           default:
           default: Vagrant insecure key detected. Vagrant will automatically replace
           default: this with a newly generated keypair for better security.
       Vagrant is configured to generate a random keypair and insert it
       onto the guest machine, but it appears Vagrant doesn't know how to do
       this with your guest OS. Please disable key insertion by setting
       `config.ssh.insert_key = false` in the Vagrantfile.

       After doing so, run `vagrant reload` for the setting to take effect.

       If you'd like Vagrant to learn how to insert keys on this OS, please
       open an issue with details about your environment.
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #create action: [Expected process to exit with [0], but received '1'
---- Begin output of vagrant up --no-provision --provider=vmware_fusion ----
STDOUT: Bringing machine 'default' up with 'vmware_fusion' provider...
==> default: Cloning VMware VM: 'mac_os_10_10_2_vmware'. This can take some time...
==> default: Verifying vmnet devices are healthy...
==> default: Preparing network adapters...
==> default: Starting the VMware VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 192.168.163.214:22
    default: SSH username: vagrant
    default: SSH auth method: private key
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
STDERR: Vagrant is configured to generate a random keypair and insert it
onto the guest machine, but it appears Vagrant doesn't know how to do
this with your guest OS. Please disable key insertion by setting
`config.ssh.insert_key = false` in the Vagrantfile.

After doing so, run `vagrant reload` for the setting to take effect.

If you'd like Vagrant to learn how to insert keys on this OS, please
open an issue with details about your environment.
---- End output of vagrant up --no-provision --provider=vmware_fusion ----
Ran vagrant up --no-provision --provider=vmware_fusion returned 1]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by bgandon** [comment]
Hi,

I'm hitting the same issue.

```
$ kitchen verify macosx
-----> Starting Kitchen (v1.3.1)
-----> Creating <default-macosx-1010>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: Importing base box 'opscode-macosx-10.10'...
==> default: Matching MAC address for NAT networking...
       ==> default: Setting the name of the VM: default-macosx-1010_default_1427780770164_59605
       ==> default: Clearing any previously set network interfaces...
       ==> default: Preparing network interfaces based on configuration...
           default: Adapter 1: nat
           default: Adapter 2: hostonly
       ==> default: Forwarding ports...
           default: 22 => 2222 (adapter 1)
       ==> default: Booting VM...
       ==> default: Waiting for machine to boot. This may take a few minutes...
           default: SSH address: 127.0.0.1:2222
           default: SSH username: vagrant
           default: SSH auth method: private key
           default: Warning: Connection timeout. Retrying...
           default: Warning: Connection timeout. Retrying...
           default: 
           default: Vagrant insecure key detected. Vagrant will automatically replace
           default: this with a newly generated keypair for better security.
       Vagrant is configured to generate a random keypair and insert it
       onto the guest machine, but it appears Vagrant doesn't know how to do
       this with your guest OS. Please disable key insertion by setting
       `config.ssh.insert_key = false` in the Vagrantfile.
       
       After doing so, run `vagrant reload` for the setting to take effect.
       
       If you'd like Vagrant to learn how to insert keys on this OS, please
       open an issue with details about your environment.
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #create action: [Expected process to exit with [0], but received '1'
---- Begin output of vagrant up --no-provision --provider=virtualbox ----
STDOUT: Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'opscode-macosx-10.10'...
==> default: Matching MAC address for NAT networking...
==> default: Setting the name of the VM: default-macosx-1010_default_1427780770164_59605
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 => 2222 (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
    default: Warning: Connection timeout. Retrying...
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
STDERR: Vagrant is configured to generate a random keypair and insert it
onto the guest machine, but it appears Vagrant doesn't know how to do
this with your guest OS. Please disable key insertion by setting
`config.ssh.insert_key = false` in the Vagrantfile.

After doing so, run `vagrant reload` for the setting to take effect.

If you'd like Vagrant to learn how to insert keys on this OS, please
open an issue with details about your environment.
---- End output of vagrant up --no-provision --provider=virtualbox ----
Ran vagrant up --no-provision --provider=virtualbox returned 1]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

I just read in the [CHANGELOG](https://github.com/mitchellh/vagrant/blob/master/CHANGELOG.md) that the upcoming 1.7.3 verison should fix it.

So, let's try to build vagrant from its source code:

```
$ git clone git@github.com:mitchellh/vagrant.git
$ cd vagrant
$ rake install
The git source https://github.com/mitchellh/vagrant-spec.git is not yet checked out. Please run `bundle install` before trying to start your application
```

Okay, so let's do it:
```
$ bundle install
Fetching https://github.com/mitchellh/vagrant-spec.git
Fetching gem metadata from https://rubygems.org/..........
Fetching version metadata from https://rubygems.org/..
Resolving dependencies...
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    vagrant (>= 0) ruby depends on
      bundler (< 1.8.0, >= 1.5.2) ruby

  Current Bundler version:
    bundler (1.8.3)

This Gemfile requires a different version of Bundler.
Perhaps you need to update Bundler by running `gem install bundler`?
```

So, let's find a [suitable version](https://github.com/bundler/bundler/blob/master/CHANGELOG.md), and install it:
```
$ gem install bundler -v1.7.13
Fetching: bundler-1.7.13.gem (100%)
Successfully installed bundler-1.7.13
Parsing documentation for bundler-1.7.13
Installing ri documentation for bundler-1.7.13
Done installing documentation for bundler after 2 seconds
1 gem installed
$ bundle install
Fetching gem metadata from https://rubygems.org/..........
Fetching version metadata from https://rubygems.org/..
Resolving dependencies...
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    vagrant (>= 0) ruby depends on
      bundler (< 1.8.0, >= 1.5.2) ruby

  Current Bundler version:
    bundler (1.8.3)

This Gemfile requires a different version of Bundler.
Perhaps you need to update Bundler by running `gem install bundler`?
```

Of course. I also have a newer version of bundler that I don't want to uninstall because I use it in another project of mine.
```
$ gem list bundler

*** LOCAL GEMS ***

bundler (1.8.3, 1.7.13)
```

But bundler can help me run the right version of bundler, right?
```
$ bundler exec bundler
Could not find gem 'webmock (~> 1.20) ruby' in the gems available on this machine.
Run `bundle install` to install missing gems.
```

Is it an issue with bundler? (I feel so much new to ruby)
```
$ rake install
Could not find gem 'webmock (~> 1.20) ruby' in the gems available on this machine.
Run `bundle install` to install missing gems.
```

Okay, let me see if I have any conflicting version:
```
$ gem list webmock

*** LOCAL GEMS ***


```
Nothing!

It just looks like I need to install all the stuff *manually*.

Let's go find a [suitable version](https://rubygems.org/gems/webmock) and install it:
```
$ gem install webmock -v1.20.4
$ bundler exec bundler
Could not find gem 'fake_ftp (~> 0.1) ruby' in the gems available on this machine.
Run `bundle install` to install missing gems.
```
And again!

Thus my question: <strong>Is is there any *automated* way to install the dependencies that are required to build vagrant?</strong>

**Comment by bgandon** [comment]
Okay, got it.

Sorry for the inconvenience.

```
$ gem install bundler -v1.7.13
$ gem uninstall bundler -v1.8.3 # the conflicting version
$ gem list bundler

*** LOCAL GEMS ***

bundler (1.7.13)
$ rake install
```
Then it's fine. Nokogiri just complained about "missing" `libiconv`, but `$ xcode-select --install` properly did the trick [as they advise](http://www.nokogiri.org/tutorials/installing_nokogiri.html#mac_os_x).

But, just an idea: <strong>Shouldn't we document this trick in the [README.md](https://github.com/mitchellh/vagrant#installing-the-gem-from-git)?</strong>

**Comment by tj** [comment]
This is really confusing, haven't had any luck with various attempts to get a base box that actually works because of these issues. Anyone have luck getting a config to work?

**Comment by rmoriz** [comment]
Same issue here with OSX 10.10 guest (build with Packer) on 1.7.2
Steps:  https://gist.github.com/rmoriz/37b671afe53c984b2f85
Source of key installation: https://github.com/timsutton/osx-vm-templates/blob/master/scripts/vagrant.sh#L13-L18

**Comment by sethvargo** [comment]
Hi all,

I believe this issue has been fixed as of Vagrant 1.7.3. Since this thread has diverged a bit from the original topic, I am going to close it and request that anyone still experiencing this open a _new_ issue and include the complete Vagrantfile as well as the output of the `up` command in [debug mode](http://docs.vagrantup.com/v2/other/debugging.html). Thanks!
